src/libraries/System.Security.Principal.Windows/tests/SecurityIdentifierTests.cs

```diff
@@ -9,6 +9,43 @@
 
 public class SecurityIdentifierTests
 {
+
+    [Fact]
+    public void SecurityIdentifierBinaryCtorThrowsOnNull()
+    {
+        Assert.Throws<ArgumentNullException>(() => new SecurityIdentifier(null, 0));
```

Comment:
Ideally, all of these would use the AssertExtensions.Throws overload that validates the argument name property.

Off the top of my head, that's

```C#
AssertExtensions.Throws<ArgumentNullException>(
    () => new SecurityIdentifier(null, 0),
    "binaryForm");
```

But I could be backwards.

Reply:
There's a overload of Throws that allows specifying the parameter name, `Assert.Throws<ArgumentNullException>("binaryForm", () => new SecurityIdentifier(null, 0));` , I can use that unless there's some reason to use the extensions one instead.

Reply:
The AssertExtensions one also has overloads for when/if the parameter name diverges between corefx/netfx.  All the places I can think of use AssertExtensions (there are lots of hits for "AssertExtensions.Throws" and mostly appear to be ArgumentException ones), so let's just be consistent and use AssertExtensions.